### PR TITLE
Add test coverage for uncovered code for rdflib serializer

### PR DESCRIPTION
**Pull request by Ja-Gk-00**
serialize.py for rdflib had important functions uncovered through tests.
Adding integration + 2 unit tests to (almost) fully cover the script and boost coverage of the codebase.

### CODE REVIEW

File: tests/integration_tests/test_rdflib/test_streams_serializer.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+
+def test_grouped_stream_to_file() -> None:
+    def _gen_graphs() -> Generator[Graph | Dataset, None, None]:
+        yield _graph_gen()
+        yield _graph_gen()
+
+    buf = io.BytesIO()
+    grouped_stream_to_file(_gen_graphs(), buf)
+    assert buf.tell() > 0
```
Comment (Ostrzyciel) [inline tests/integration_tests/test_rdflib/test_streams_serializer.py:84]: In a few tests I see there that you are just checking if the buffer is non-empty – please update it so that we check if after parsing, it really contains what we expect it to contain.

One thing is coverage, another is "is this testing the thing that we want to be tested?" :)

File: tests/integration_tests/test_examples/temp/temp_output.jelly
Comment (Ostrzyciel) [inline tests/integration_tests/test_examples/temp/temp_output.jelly:1]: Please add a rule to gitignore to exclude this in the future. You can place a gitignore in the test_exampled directory with rule `/temp`

**Review by Ostrzyciel — CHANGES_REQUESTED** [review]
2 minor things

File: tests/integration_tests/test_rdflib/test_streams_serializer.py (hunk 1 above)
Comment (Ja-Gk-00) [reply tests/integration_tests/test_rdflib/test_streams_serializer.py]: Ok, added more robust testing, now we actually check if what is in is out, not just if stuff is non empty.
